pkg/client/dtclient/settings_client.go

```diff
@@ -563,24 +563,48 @@ func doObjectsMatchBasedOnUniqueKeys(uniqueKeys []string, source SettingsObject,
 	return true, matches, nil
 }
 
-func isSameValueForKey(key string, c1 []byte, c2 []byte) (same bool, matchingVal any, err error) {
+func isSameValueForKey(targetPath string, c1 []byte, c2 []byte) (same bool, matchingVal any, err error) {
 	u := make(map[string]any)
```

Comment:
Could you please call this.... something else? :)
`UnmarshaledConfig`? Better names are welcome